File: leetcode2/1easy/최원준/Q3184.java
package Leetcode.최원준;

/*
1. 아이디어 :
이중 포문으로 모든 조합을 확인

2. 시간복잡도 :
O( n*n)

3. 자료구조/알고리즘 :
- / 브루트 포스
 */

public class Q3184 {
    class Solution {
        public int countCompleteDayPairs(int[] hours) {
            int ans = 0;
            int n = hours.length;

            for (int i=0; i<n; i++) for (int j=i+1; j<n; j++) {
                if ((hours[i] + hours[j])%24==0) ans++;
            }
            return ans;
        }
    }
}

File: leetcode2/1easy/강병호/Q3184.py
class Solution:
    def countCompleteDayPairs(self, hours: list[int]) -> int:
        # 24의 배수 구하기
        # % 24 == 0 
        result = 0

        for i in range(0, len(hours)):
            for j in range(i+1, len(hours)):
                num = hours[i] + hours[j]
                if num % 24 == 0:
                    result += 1

        
        return result


if __name__ == '__main__':
    sol = Solution()

    print(sol.countCompleteDayPairs([12,12,30,24,24]))
    print(sol.countCompleteDayPairs([72,48,24,3]))